TESTS/mbed_drivers/stl_features/main.cpp

```diff
@@ -59,62 +54,49 @@ void BubbleSort(T& _array, size_t array_size, F functor) {
     }
 }
 
-struct printInt {
-    void operator()(int i) { printf("%d ", i); }
-};
-
-struct printFloat {
-    void operator()(float f) { printf("%f ", f); }
-};
-
-struct printString {
-    void operator()(const char* s) { printf("%s ", s); }
-};
-
 struct greaterAbs {
     bool operator()(int a, int b) { return abs(a) > abs(b); }
 };
 
 }   // namespace
 
 void test_case_stl_equal() {
-    std::vector<int> v_pints(p_integers, p_integers + TABLE_SIZE(p_integers));
-    TEST_ASSERT_TRUE(std::equal(v_pints.begin(), v_pints.end(), p_integers));
+    const int n_integers[] = {NEGATIVE_INTEGERS};
+    std::vector<int> v_pints(n_integers, n_integers + TABLE_SIZE(n_integers));
+    TEST_ASSERT_TRUE(std::equal(v_pints.begin(), v_pints.end(), n_integers));
 }
 
 void test_case_stl_transform() {
+    const float floats[] = {FLOATS};
     const char* floats_str[] = {FLOATS_STR};
     float floats_transform[TABLE_SIZE(floats_str)] = {0.0};
     std::transform(floats_str, floats_str + TABLE_SIZE(floats_str), floats_transform, atof);
-    //printf("stl_transform::floats_str: ");
-    //std::for_each(floats_str, floats_str + TABLE_SIZE(floats_str), printString());
-    //printf("stl_transform::floats_transform: ");
-    //std::for_each(floats_transform, floats_transform + TABLE_SIZE(floats_transform), printFloat());
-    //printf("\n");
 
     TEST_ASSERT_TRUE(std::equal(floats_transform, floats_transform + TABLE_SIZE(floats_transform), floats));
 }
 
 void test_case_stl_sort_greater() {
```

Comment:
I hope I'm not repeating a discussion, but have we considered if it's worthwhile to keep these tests?

You could spend a lifetime trying to add coverate to the stl....